### DISCUSSION TITLE
Welcome to floor-heating-proportional-valve Discussions!

### DISCUSSION THREAD

**Discussion by nliaudat** [nliaudat/floor-heating-proportional-valve#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by fabianmuehlberger** [comment]
Hi, 
I recently started working on an open source smart radiator valve and someone on the ESPHome discord pointed out, that you are working on one. I already did some designing and testing, but I am currently in design and evaluation phase. If you are interested in collaborating to this idea, let me know. 
From what I can see, your project and components are very similar, if you like I would love to exchange design ideas, validate and compare. 
https://github.com/fabianmuehlberger/OpenValve
Best Fabian

**Reply by nliaudat** [reply]
Hi, 

Already, thanks for your support of open-source/hardware. 

I've had a quick look at your project and I see several problems I've had to deal with: 

1) The concept of an M3 fine-pitch motor with reduction gearbox seems correct on paper, but in use I had to add a spring to compensate for the extra pressure of the heating circuit in winter. The forces involved can be as high as 150N.

2) Your 3D-printed housing will never withstand the torsional forces generated by the motor. 

3) The price of your components (motor + polulu + esp) is much higher than mine (~$12).

4) You need to think about battery power in the 1st design phase (I didn't).

5) hall effect encoder cannot store absolute position in case of power fail. That's why I have optical position sensor

I'm not going to join your project, as mine is much more advanced and functional. However, I'll be happy to answer any questions you may have. 

I wish you good luck!

**Reply by Tecnologic** [reply]
Hi Nicolas,

Why do u think that a absolute position feedback is needed. Right now I'm running the 8ch controller with vdmot drives and the only problem I have is the unreliable endstops.

I designed a spin off from Ur PCB with real in phase current sense and want to run the drives like a industrial servo. Position feedback I get from the communication spikes. That gives me a incremental encoder. So the powerloss procedure would be closing the valve on boot up to find the endstop and then  0 the encoder counts.

Do U see any drawbacks on this approach?

Regards

Alex

**Reply by fabianmuehlberger** [reply]
Hey, 
hope you're making progress in your development, and thank you for your feedback. 
I had similar issues with my design and the n20 300/1 gearbox motor not having a fun time with 3.3 V. 
After a long time on the interwebs I finally found a motor specifically made for this application, it's a canned stepper with a 10/1 gearbox and linear travel of 5mm. For ~ 6 euro a piece.
If you are interested, I let you know when they arrive, and I had a chance of testing them.

Best 
Fabian

**Reply by nliaudat** [reply]
Hi Alex,
Stepper motors are more complicated to run and cost a lot.
I've seen valve actuators which use a traditional motor and an opto encoder to achieve the same result.
Absolute position feedback is not needed, but for the price of a simple opto sensor you can get absolute position as the total movement is less than 5mm.

I have tested all the following solutions 
![image](https://github.com/nliaudat/floor-heating-proportional-valve/assets/6782613/e5925029-9693-4d56-b778-3cc46c37af7e)
and the best is opto sensor as there is no contact, so no wear nor oxidation.

You can also make an accurate DC motor with opto sensors : https://www.hackster.io/adamb314/super-accurate-300-robot-arm-15f1d2

Regards

**Reply by Tecnologic** [reply]
I want to keep it hardware wise as simple as possible. The current waveform of the brushed DC motor of the vdmot valve gives a spike at every communication of the brushed. I count these spikes to have an incremental encoder. To reference the drive I use the closed position. Just need to integrate that in esphome and test it with my heating valve array.

**Reply by nliaudat** [reply]
The question is : do we really need precision in the position of the valve? My philosophy in central heating solution is to generate small movements of the valve according to the temperature difference in order to obtain the ideal level by increments with a fairly short time interval. And it works! In my opinion, the information on the endstops is sufficient, But if you're going to create a new valve actuator, that's well to include an absolute position at a moderate cost.

**Reply by Tecnologic** [reply]
I'm totally with U there. That's why I asked about the absolute position measurement/battery need for the current implementation. On my desk with a valve without water my implementation works as I want it. There may be slight drift in the absolute position due to start/stop of the motor. But that can be fixed easily by cyclic reference after X moves or a time interval.

**Comment by nliaudat** [comment]
I've tested the valve with https://github.com/nliaudat/esp32_8ch_motor_shield and it's working with additionnal spring on axle.

![image](https://github.com/nliaudat/floor-heating-proportional-valve/assets/6782613/49ab7794-1c02-4cbc-aa20-e2002b8697c0)

**Comment by Benas09** [comment]
Hi. I started to solve the same problem (proportional heating actuator) and designed and printed a prototype which is very similar to yours. Then found this project and I saw that you already solved it :)

I thought to use hall sensor. I see you tried it also. Is infrared more accurate and stable than hall, or why did you choose infrared?

Also I found that you have a controller for underfloor heating. If that controller is used, then PCB's with esp for every actuator is not needed?

![Screenshot_20260116_230948_Onshape](https://github.com/user-attachments/assets/cb3f9736-caf2-4432-9bed-1fc56e974d38)

**Reply by nliaudat** [reply]
Hi, 
1) I used the infrared sensors cause it was the best in precision/price/complexity
2) The main controller board is actually what I use cause it power off the valves. 
The esp board is used only on battery mode that is not implemented.

**Reply by Benas09** [reply]
Thank you for response.

So, if I want to use your controller, I have to order pcbs for each actuator, but do not place esps in them?
Is FDM printed parts are okay? I guess that transparent parts I saw are resin printed.
I see this project is several years old now - what is expected life of these actuators?
What accuracy is expected with these boards using optocouplers? I mean, 5% steps or 10% steps are possible?

**Reply by nliaudat** [reply]
If you want to use the main controller, there is no need of pcb at all. (I made https://github.com/nliaudat/floor-heating-proportional-valve/tree/main/pcb/valve_only cause soldering wires on the motor was difficult.)

With main controller, the project is only motorized valve and do not use optocouplers. The endstop are detected with ADC BEMF of motor.

The precision of ITR8307 depends on many factors : 

<img width="348" height="331" alt="image" src="https://github.com/user-attachments/assets/56fb9b53-3e8c-4397-aa9e-beb4f3010e24" />

-The reflection properties of the detected material (I glue aluminium foil at top of slilder for better reflection)
-The ADC reading precision of your board
-The moving speed and ADC measurment frequency

With all parameters in green, you can detect steps of 0.10 mm.

The transparent resin is just a demonstration to see inside of valve. With PLA or ABS, the strengh and longevity is very good. 
The only part that is less resistant is the motor. It miss a bit of torque
